### PR TITLE
SkyWalking Agent Support

### PR DESCRIPTION
**Pull request by arghya88**
This change updates the buildpack to include SkyWalking agent support.

Please see http://skywalking.io for more details

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ The buildpack supports extension through the use of Git repository forking. The
   * [Google Stackdriver Debugger](docs/framework-google_stackdriver_debugger.md) ([Configuration](docs/framework-google_stackdriver_debugger.md#configuration))
   * [Introscope Agent](docs/framework-introscope_agent.md) ([Configuration](docs/framework-introscope_agent.md#configuration))
   * [JaCoCo Agent](docs/framework-jacoco_agent.md) ([Configuration](docs/framework-jacoco_agent.md#configuration))
+  * [SkyWalking Agent](docs/framework-sky_walking_agent.md) ([Configuration](docs/framework-sky_walking_agent.md#configuration))
```
Comment (nebhale) [inline README.md]: This should be sorted alphabetically.

File: docs/framework-sky_walking_agent.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+For general information on configuring the buildpack, including how to specify configuration values through environment variables, refer to [Configuration and Extension][].
+
+The framework can be configured by modifying the [`config/sky_walking_agent.yml`][] file in the buildpack fork. The framework uses the [`Repository` utility support][repositories] and so it supports the [version syntax][] defined there.
+
+| Name | Description
+| ---- | -----------
+| `default_application_name` | This is omitted by default but can be added to specify the application name in the SkyWalking dashboard. This can be overridden by an `application-name` entry in the credentials payload. If neither are supplied the default is the `application_name` as specified by Cloud Foundry.
+| `repository_root` | The URL of the SkyWalking repository index ([details][repositories]).
+| `version` | The version of SkyWalking to use. Candidate versions can be found in [this listing][].
+| `enabled` | Whether to enable the SkyWalking Agent. By default the agent is turned off.
```
Comment (nebhale) [inline docs/framework-sky_walking_agent.md]: This is unnecessary as the existence of a service bound to the application determines if it is enabled or disabled.

File: config/sky_walking_agent.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+
+# Configuration for the Spring Auto Reconfiguration framework.
+# Note that the repository is shared with the Play Auto Reconfiguration framework and should be kept in step to
+# avoid conflicts.
+---
+version: 3.+
+default_application_name: $(ruby -e "require 'json' ; a = JSON.parse(ENV['VCAP_APPLICATION']);
+  puts \"#{a['space_name']}:#{a['application_name']}\"")
+repository_root: https://raw.githubusercontent.com/apache/incubator-skywalking/releases/download/v3.2.6/skywalking-agent.tar.gz
```
[the remark this answers is not in the document]
Comment (arghya88) [reply config/sky_walking_agent.yml]: I don't think they have hosted it. Please host it in Cloud Foundry repository

[the remark this answers is not in the document]
Comment (nebhale) [reply config/sky_walking_agent.yml]: Then that will delay this integration for a while.

**Comment by nebhale** [comment]
I've added the dependency to the repository.  Can you please test using the buildpack (either `-b` or `buildpack` manifest entry) `https://github.com/cloudfoundry/java-buildpack.git#556-skywalking`.

**Comment by arghya88** [comment]
screen shot 2018-02-27 at 7 33 36 pm

**Pull request closed without merging**